src/mlpack/core/data/split_data.hpp

```diff
@@ -179,50 +155,43 @@ void StratifiedSplit(const arma::Mat<T>& input,
  * @param shuffleData If true, the sample order is shuffled; otherwise, each
  *       sample is visited in linear order. (Default true.)
  */
-template<typename T, typename U>
+template<typename T, typename LabelsType,
+         typename = std::enable_if_t<arma::is_arma_type<LabelsType>::value> >
 void Split(const arma::Mat<T>& input,
-           const arma::Row<U>& inputLabel,
+           const LabelsType& inputLabel,
            arma::Mat<T>& trainData,
            arma::Mat<T>& testData,
-           arma::Row<U>& trainLabel,
-           arma::Row<U>& testLabel,
+           LabelsType& trainLabel,
+           LabelsType& testLabel,
            const double testRatio,
            const bool shuffleData = true)
 {
   const size_t testSize = static_cast<size_t>(input.n_cols * testRatio);
   const size_t trainSize = input.n_cols - testSize;
   trainData.set_size(input.n_rows, trainSize);
   testData.set_size(input.n_rows, testSize);
-  trainLabel.set_size(trainSize);
-  testLabel.set_size(testSize);
 
+  arma::uvec order = arma::linspace<arma::uvec>(0, input.n_cols - 1,
+      input.n_cols);
```

Comment:
I know that it would be more code, but initializing `order` even when `shuffleData` is `false` can cause a good bit of unnecessary RAM usage.  Personally I think it might be better to avoid using `order` at all when `shuffleData` is `false`.

(I hope this doesn't contradict comments I made previously while reviewing.  I can't remember, it's been a while.  So, sorry if I said one thing then and another now...)

Reply:
You are correct. Small optimizations eventually add up to make impact. Fixed this also in the latest patch :-)